inbc-program/tests/unit/test_source_app_parser.py

```diff
@@ -46,64 +46,64 @@ def test_parse_add_arguments_deb822_format_separate_lines_successfully(self):
+                 '--gpgKeyUri', 'https://repositories.intel.com/gpu/intel-graphics.key',
+                 '--sources', 'deb http://example.com/ focal main restricted universe',
+                 'deb-src http://example.com/ focal-security main',
+                 '--filename', 'intel-gpu-jammy.list'])
+            with pytest.raises(InbcException,
+                               match="Source requires either both gpgKeyUri and gpgKeyName "
+                                     "to be provided, or neither of them."):
+                Inbc(p, 'source', False)
+
+    def test_raise_application_add_with_only_one_gpgKeyName_param(self):
+        with patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect') as m_connect:
+            p = self.arg_parser.parse_args(
+                ['source', 'application', 'add',
+                 '--gpgKeyName', 'intel-graphics.gpg',
+                 '--sources', 'deb http://example.com/ focal main restricted universe',
+                 'deb-src http://example.com/ focal-security main',
+                 '--filename', 'intel-gpu-jammy.list'])
+            with pytest.raises(InbcException,
+                               match="Source requires either both gpgKeyUri and gpgKeyName "
+                                     "to be provided, or neither of them."):
+                Inbc(p, 'source', False)
+
+    def test_create_add_deb_822_format_manifest_successfully(self):
         p = self.arg_parser.parse_args(
             ['source', 'application', 'add',
              '--sources', 'X-Repolib-Name: Google Chrome',
-                          'Enabled: yes',
-                          'Types: deb',
-                          'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
-                          'Suites: stable',
-                          'Components: main',
+             'Enabled: yes',
+             'Types: deb',
+             'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
+             'Suites: stable',
+             'Components: main',
              '--filename', 'google-chrome.sources'])
-        Inbc(p, 'source', False)
+        with patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect') as m_connect:
+            Inbc(p, 'source', False)
         expected = '<?xml version="1.0" encoding="utf-8"?><manifest><type>source</type><applicationSource>' \
                    '<add><repo><repos><source_pkg>X-Repolib-Name: Google Chrome</source_pkg>' \
                    '<source_pkg>Enabled: yes</source_pkg>' \
-                   '<source_pkg>Types: deb</source_pkg>'\
+                   '<source_pkg>Types: deb</source_pkg>' \
                    '<source_pkg>URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub</source_pkg>' \
                    '<source_pkg>Suites: stable</source_pkg>' \
                    '<source_pkg>Components: main</source_pkg>' \
                    '</repos><filename>google-chrome.sources</filename></repo></add></applicationSource></manifest>'
```

Comment:
**Suggestion:** Instead of manually constructing XML strings, consider using an XML library to build and compare XML documents, which would be less error-prone and more maintainable. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        from xml.etree.ElementTree import Element, SubElement, tostring
        manifest = Element('manifest')
        applicationSource = SubElement(manifest, 'applicationSource')
        # ... construct the rest of the XML structure ...
        expected = tostring(manifest).decode()
```